src/process_util.rs

```diff
@@ -135,8 +153,11 @@ pub fn run_and_wait(cmd: &mut Command)
                         Stop(..) | Continue(..) | Death(..) => { }
                     }
                 }
-                try!(tty_guard.check().map_err(|e|
-                    format!("Error handling tty: {}", e)));
+                if take_tty {
+                    try!(tty_guard.check().map_err(|e|
+                        format!("Error handling tty: {}", e)));
+
+                }
             }
             _ => unreachable!(),
```

Comment:
> SIGTTOU is only generated for an attempt to write to the terminal if the TOSTOP output mode is set; see Output Modes.

This is why SIGTTOU is not generated

> It is very interesting that vagga launcher received SIGTTIN

Probably `less` tried to read and whole process group was sent SIGTTIN to

> Should we ignore those signals or propagate SIGTSTP to children and send SIGSTOP self?

It looks like we should stop all children:
```
> bash | less &
[1] 12057 12058
[1]  + suspended (tty output)  bash | less
```